launcher/internal/gpu/driverinstaller.go

```diff
@@ -0,0 +1,169 @@
+	deviceinfo.H100,
+}
+
+// DriverInstaller contains information about the gpu driver installer settings
+type DriverInstaller struct {
+	cdClient   *containerd.Client
+	launchSpec spec.LaunchSpec
+	logger     *log.Logger
+}
+
+// NewDriverInstaller instanciates an object of driver installer
+func NewDriverInstaller(cdClient *containerd.Client, launchSpec spec.LaunchSpec, logger *log.Logger) *DriverInstaller {
+	return &DriverInstaller{
+		cdClient:   cdClient,
+		launchSpec: launchSpec,
+		logger:     logger,
+	}
+}
+
+// InstallGPUDrivers installs the GPU driver on host machine using the cos-gpu-installer container.
+// This function performs the same steps specified in this README file:
+// https://pkg.go.dev/cos.googlesource.com/cos/tools.git@v0.0.0-20241008015903-8431fe581b1f/src/cmd/cos_gpu_installer#section-readme
+// README specifies docker command where this function uses containerd for launching and managing the gpu driver installer container.
+func (di *DriverInstaller) InstallGPUDrivers(ctx context.Context) error {
+	err := remountAsExecutable(InstallationHostDir)
+	if err != nil {
+		return fmt.Errorf("failed to remount the installation directory: %v", err)
+	}
+
+	gpuType, err := deviceinfo.GetGPUTypeInfo()
+	if err != nil {
+		return fmt.Errorf("failed to get the gpu type info: %v", err)
+	}
+
+	if !gpuType.OpenSupported() {
+		return fmt.Errorf("unsupported gpu type %s, please retry with one of the supported gpu types: %v", gpuType.String(), supportedGpuTypes)
+	}
+
+	ctx = namespaces.WithNamespace(ctx, namespaces.Default)
+	installerImageRef, err := getInstallerImageReference()
+	if err != nil {
+		di.logger.Printf("failed to get the installer container image reference: %v", err)
+		return err
+	}
+
+	di.logger.Printf("cos gpu installer version : %s", installerImageRef)
+	image, err := di.cdClient.Pull(ctx, installerImageRef, containerd.WithPullUnpack)
+	if err != nil {
+		return fmt.Errorf("failed to pull installer image: %v", err)
+	}
+
+	mounts := []specs.Mount{
+		{
+			Type:        "volume",
+			Source:      "/dev",
+			Destination: "/dev",
+			Options:     []string{"rbind", "rw"},
+		}, {
+			Type:        "volume",
+			Source:      "/",
+			Destination: "/root",
+			Options:     []string{"rbind", "rw"},
+		},
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		di.logger.Printf("cannot get hostname: %v", err)
+	}
+
+	container, err := di.cdClient.NewContainer(
+		ctx,
+		installerContainerID,
+		containerd.WithImage(image),
+		containerd.WithNewSnapshot(installerSnapshotID, image),
+		containerd.WithNewSpec(oci.WithImageConfig(image),
+			oci.WithPrivileged,
+			oci.WithProcessArgs("/cos-gpu-installer", "install", "-version=default", fmt.Sprintf("-host-dir=%s", InstallationHostDir)),
+			oci.WithAllDevicesAllowed,
+			oci.WithHostDevices,
+			oci.WithMounts(mounts),
+			oci.WithHostNamespace(specs.NetworkNamespace),
+			oci.WithHostNamespace(specs.PIDNamespace),
+			oci.WithHostHostsFile,
+			oci.WithHostResolvconf,
```

Comment:
since a lot of this is duplicated, it would be useful to have helper functions that help mount with hostnamespace, hosts file and resolve conf.